### PR TITLE
Minor Utils refactoring and testing

### PR DESCRIPTION
**Pull request by ftomassetti**
Refactor and test Utils.capitalize e decapitalize

### CODE REVIEW

File: javaparser-core/src/main/java/com/github/javaparser/utils/Utils.java [hunk 1]
```diff
@@ -147,35 +149,39 @@ public static String camelCaseToScreaming(String input) {
...
+    public static String decapitalize(String s) {
+        return stringTransformer(s, it -> it.toLowerCase());
+    }
+
+    private static String stringTransformer(String s, Function<String, String> transformation) {
+        if (s.isEmpty()) {
+            throw new IllegalArgumentException();
         }
-        return string.substring(0, index);
+        StringBuffer sb = new StringBuffer();
```
Comment (matozoid) [inline javaparser-core/src/main/java/com/github/javaparser/utils/Utils.java]: StringBuilder please!

File: javaparser-core/src/main/java/com/github/javaparser/utils/Utils.java [hunk 2]
```diff
@@ -147,35 +149,39 @@ public static String camelCaseToScreaming(String input) {
...
-        int index = 0;
-        while (index < string.length() && !Character.isWhitespace(string.charAt(index))) {
-            index++;
+    public static String decapitalize(String s) {
+        return stringTransformer(s, it -> it.toLowerCase());
+    }
+
+    private static String stringTransformer(String s, Function<String, String> transformation) {
+        if (s.isEmpty()) {
+            throw new IllegalArgumentException();
```
Comment (matozoid) [inline javaparser-core/src/main/java/com/github/javaparser/utils/Utils.java]: Add a little message to the exception.

File: javaparser-core/src/main/java/com/github/javaparser/utils/Utils.java [hunk 3]
```diff
@@ -147,35 +149,39 @@ public static String camelCaseToScreaming(String input) {
...
         }
-        return string.substring(0, index);
+        StringBuffer sb = new StringBuffer();
+        sb.append(transformation.apply(s.substring(0, 1)));
+        if (s.length() > 1) {
+            sb.append(s.substring(1));
```
Comment (matozoid) [inline javaparser-core/src/main/java/com/github/javaparser/utils/Utils.java]: I think the "if" is superfluous.

File: javaparser-testing/src/test/java/com/github/javaparser/utils/UtilsTest.java [hunk 4]
```diff
@@ -42,4 +42,34 @@ public void screamingEmptyString() {
...
+
+    @Test
+    public void capitalizeOnStringOfTwoCharacters() {
+        assertEquals("Fo", capitalize("fo"));
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void decapitalizeOnEmptyString() {
+        decapitalize("");
+    }
```
Comment (matozoid) [inline javaparser-testing/src/test/java/com/github/javaparser/utils/UtilsTest.java:64]: Thanks for the unit tests!

**Comment by ftomassetti** [comment]
Corrections applied & rebased

**Pull request merged**